### PR TITLE
adding option for excel conditional formating [sc-161134]

### PR DESCRIPTION
**Pull request by liamlynch-data**
Pretty straightforward 
(assuming people are happy with the DSS backend changes for this)

Needs this branch of DSS to work (till we merge it)
https://github.com/dataiku/dip/tree/feature/dss12-sc-161134-apply-coloring-for-api-for-plugin

### CODE REVIEW

**Comment by liamlynch-data** [comment]
I just thought - probably we don't want the new option to show unless we're on a version of DSS that supports it. I will try to add something to do that now.

**Comment by liamlynch-data** [comment]
> I just thought - probably we don't want the new option to show unless we're on a version of DSS that supports it. I will try to add something to do that now.

Done in https://github.com/dataiku/dss-plugin-sendmail/pull/10/commits/1bc91fa2f89e76dae578cb7b94202f5d11d1aafa

**Review by alexbourret — APPROVED** [review]
All good !

File: resource/dynamic_form.py [hunk 1]
```diff
@@ -1,15 +1,20 @@
 from dss_selector_choices import DSSSelectorChoices, SENDER_SUFFIX
 import dataiku
 
-def do(payload, config, plugin_config, inputs):
 
+def is_12_6_plus(dss_client):
```
Comment (MayeulRousselet) [inline resource/dynamic_form.py]: Better to rename it to `support_messaging_channel_and_conditionnal_formatting`

File: python-lib/dku_attachment_handling.py [hunk 2]
```diff
@@ -21,24 +21,29 @@ def attachments_template_dict(attachment_datasets):
...
+        :param apply_coloring_excel: boolean, whether to apply conditional formatting (aka coloring) for Excel attachments
         :return: Attachments as List of AttachmentFile
     """
+    logging.info(f"Building attachments, type: {attachment_type}, apply colouring? {apply_coloring_excel}")
+    is_excel = attachment_type == "excel" or attachment_type == "excel_can_ac"
 
-    if attachment_type == "excel":
+    if is_excel:
         request_fmt = "excel"
+        format_params = {"applyColoring": (apply_coloring_excel and attachment_type == "excel_can_ac")}
```
Comment (MayeulRousselet) [inline python-lib/dku_attachment_handling.py]: `applyColoring` will be sent on DSS < 12.6 (not a big deal)

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]

**Review by MayeulRousselet — COMMENTED** [review]
Tested it with attaching foreign dataset that have conditional formatting on the source:
- Expected: no CF is applied
- Actual: CF is applied.

this is due to an issue in the export (https://app.shortcut.com/dataiku/story/173249) not to be covered here so.

File: resource/dynamic_form.py (hunk 1 above)
Comment (liamlynch-data) [reply resource/dynamic_form.py]: Renamed [here](https://github.com/dataiku/dss-plugin-sendmail/pull/10/commits/052303cae4f8c86783dcd9c7359c7d255d622f48#diff-44f227161ef56923c43a70ade7493cd01df31e6e43d33fba892abba4021da9b4)

File: python-lib/dku_attachment_handling.py (hunk 2 above)
Comment (liamlynch-data) [reply python-lib/dku_attachment_handling.py]: Changed so only sends format_params where needed [here ](https://github.com/dataiku/dss-plugin-sendmail/pull/10/commits/052303cae4f8c86783dcd9c7359c7d255d622f48#diff-e84ac22d39938fd532a4d10e767fc0e6f64cdc3b98225263cedcf204d6ff6865)

**Review by MayeulRousselet — APPROVED** [review]
LGTM thx for the changes

**Pull request merged**